src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/controller/ApiKeyController.java

```diff
@@ -82,6 +82,20 @@
         return result;
     }
 
+    /**
+     * update an existing api key
+     */
+    @RequestMapping(value = "/{id}", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<PublicKeyData> updateKey(@PathVariable("id") long id,
+                                                          @RequestBody List<String> roles) {
+        try {
+            PersistentApiKeyType key = apiKeyRepository.update(id, roles);
+            return new ResponseEntity<>(new PublicKeyData(key.getId(), key.getRoles()), HttpStatus.OK);
+        } catch (InvalidRoleNameException | ApiKeyRepository.ApiKeyNotFoundException e) {
```

Comment:
ApiKeyNotFoundException should probably result in ResourceNotFoundException instead (see `revoke`)

Reply:
Fixed.